tests/test_pdf_qgauss.py

```diff
@@ -0,0 +1,60 @@
+#  Copyright (c) 2024 zfit
+import numpy as np
+import pytest
+import zfit
+from zfit import z
+import zfit.z.numpy as znp
+from numba_stats import qgaussian as qgaussian_numba
+import tensorflow as tf
+
+mu_true = 0.5
+sigma_true = 1.1
+true_integral_dict = {1.00001: 0.9999782, 1.5: 0.9695, 2.0: 0.8063152, 2.5: 0.4805167, 2.9: 0.1078687, 2.99: 0.011044}
+
+
+def create_qgauss(q, mu, sigma, limits):
+    obs = zfit.Space("obs1", limits)
+    qgauss = zfit.pdf.QGauss(q=q, mu=mu, sigma=sigma, obs=obs)
+    return qgauss, obs
+
+
+@pytest.mark.parametrize("q", [1.00001, 1.5, 2.0, 2.5, 2.9, 2.99])
+def test_qgauss_pdf(q):
+    qgauss, _ = create_qgauss(q=q, mu=mu_true, sigma=sigma_true, limits=(-5, 5))
+    assert qgauss.pdf(0.5, norm=False).numpy().item() == pytest.approx(
+        qgaussian_numba.pdf(0.5, q=q, mu=mu_true, sigma=sigma_true), rel=1e-5
+    )
+    np.testing.assert_allclose(
+        qgauss.pdf(tf.range(-5, 5, 10_000), norm=False).numpy(),
+        qgaussian_numba.pdf(tf.range(-5, 5, 10_000), q=q, mu=mu_true, sigma=sigma_true),
+        rtol=1e-5,
+    )
+    assert qgauss.pdf(tf.range(-5, 5, 10_000), norm=False) <= qgauss.pdf(0.5, norm=False)
+
+    sample = qgauss.sample(1000)
+    tf.debugging.assert_all_finite(sample.value(), "Some samples from the qgauss PDF are NaN or infinite")
```

Comment:
Let's rather use `assert all(np.isfinite)`, I think that the tf doesn't really create an assert assertion but throws an error. Pytest testing should use an "assert" to best recognize tests and failures